system/service.py

```diff
@@ -520,13 +520,15 @@ def sysv_is_enabled(name):
             return False
 
     def get_systemd_status_dict(self):
-
         # Check status first as show will not fail if service does not exist
         (rc, out, err) = self.execute_command("%s show '%s'" % (self.enable_cmd, self.__systemd_unit,))
-        if rc != 0:
-            self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
-        elif 'LoadState=not-found' in out:
-            self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        if self.module.params['must_exist']:
+            if rc != 0:
+                self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
+            elif 'LoadState=not-found' in out:
+                self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        else:
+            self.module.exit_json(changed=False, exists=False)
```

Comment:
Huh, I guess I didn't follow the logic.

So if `must_exist` is set, and the exit is non-zero, you get the failure from line 527. Since failure to find the service always results in non-zero, how would line 529 ever be reached?

Lets walk through when `must_exist` is set to false, it looks like the code immediately jumps to line 531, regardless of the exit code. How would this actually work if `must_exist` is set to false, but the service actually does exist?